core/services/gas_updater.go

```diff
@@ -57,91 +70,292 @@ type gasUpdater struct {
 	// it too early results in an empty block.
 	blockDelay int64
 	percentile int
+	mb         *utils.Mailbox
+	wg         *sync.WaitGroup
+	ctx        context.Context
+	ctxCancel  context.CancelFunc
+
+	logger *logger.Logger
+}
+
+type blockInternal struct {
+	Number       string
+	Hash         common.Hash
+	ParentHash   common.Hash
+	Transactions []types.Transaction
+}
+
+func toBlockNumStr(n int64) string {
+	return hexutil.EncodeBig(big.NewInt(n))
+}
+
+// Block represents an ethereum block
+type Block struct {
+	Number       int64
+	Hash         common.Hash
+	ParentHash   common.Hash
+	Transactions []types.Transaction
+}
+
+// MarshalJSON implements json marshalling for Block
+func (b Block) MarshalJSON() ([]byte, error) {
+	return json.Marshal(blockInternal{
+		toBlockNumStr(b.Number),
+		b.Hash,
+		b.ParentHash,
+		b.Transactions,
+	})
+}
+
+// UnmarshalJSON unmarshals to a Block
+func (b *Block) UnmarshalJSON(data []byte) error {
+	bi := blockInternal{}
+	if err := json.Unmarshal(data, &bi); err != nil {
+		return err
+	}
+	n, err := hexutil.DecodeBig(bi.Number)
+	if err != nil {
+		return err
+	}
+	*b = Block{
+		n.Int64(),
+		bi.Hash,
+		bi.ParentHash,
+		bi.Transactions,
+	}
+	return nil
 }
 
 // NewGasUpdater returns a new gas updater.
 func NewGasUpdater(store *store.Store) GasUpdater {
```

Comment:
it looks like all this needs is an ethClient and config, not the whole store. Can we just pass in those two things to make it explicit? Also I think if we do that then we can avoid the cltest.NewStore in the tests and just mock the config and ethClient directly to avoid db load